client/src/app/pages/applications/components/application-form/application-form.tsx

```diff
@@ -166,25 +162,27 @@ export const useApplicationFormHook = ({
       comments: string()
         .trim()
         .max(250, t("validation.maxLength", { length: 250 })),
+
+      // source code fields
+      kind: string().oneOf(["git", "subversion"]),
```

Comment:
@sjd78 
It seems that the validation is too restrictive right now - you cannot create an app without a source repo.
Although the repository is marked as required in the form it's effectively optional as you can decide to use only binaries.

Reply:
Fixed finally by rearranging the schema and using the value of the kind field to indicate if the repository field is required.